shell/components/nav/WindowManager/ContainerShell.vue

```diff
@@ -282,16 +278,44 @@ export default {
       this.socket.addEventListener(EVENT_DISCONNECTED, (e) => {
         this.isOpen = false;
         this.isOpening = false;
+
+        // If we had an error message, try connecting with the next command
+        if (this.errorMsg) {
+          this.terminal.write(this.errorMsg);
+          if (this.backupShells.length && this.retries < 2) {
+            this.retries++;
+            // we're not really counting on this being a reactive change so there's no need to fire the whole action
```

Comment:
Messages like this, in combination with the fact that `pod` is a props, lead me to believe that there might be a better pattern to utilize here. Although, I won't be too much of a stickler on it because we've long-established that we're fine with mutating object props throughout the Dashboard codebase 🙂

Reply:
Yeah, I went back and forth on this as well. Since it's not a mutation that we need to send to the API, none of the existing action/mutation pairs really do what we want here and it felt like overkill to make custom ones so I'm really just using the setter on the resource (os is a getter/setter pair using _os under the hood) which is the happiest medium in could think of.